dev/tasks/utils/git.js

```
@@ -0,0 +1,154 @@
+	 * @param {String} branchName Name of the branch to checkout.
+	 */
+	checkout( repositoryLocation, branchName ) {
+		const checkoutCommands = [
+			`cd ${ repositoryLocation }`,
+			`git checkout ${ branchName }`
+		];
+
+		tools.shExec( checkoutCommands.join( ' && ' ) );
+	},
+
+	/**
+	 * Pulls specified branch from origin.
+	 *
+	 * @param {String} repositoryLocation Absolute path to repository.
+	 * @param {String} branchName Branch name to pull.
+	 */
+	pull( repositoryLocation, branchName ) {
+		const checkoutCommands = [
+			`cd ${ repositoryLocation }`,
+			`git pull origin ${ branchName }`
+		];
+
+		tools.shExec( checkoutCommands.join( ' && ' ) );
+	},
+
+	/**
+	 * Initializes new repository, adds and merges CKEditor5 boilerplate project.
+	 *
+	 * @param {String} repositoryPath Absolute path where repository should be created.
+	 */
+	initializeRepository( repositoryPath ) {
+		const initializeCommands = [
+			`git init ${ repositoryPath }`,
+			`cd ${ repositoryPath }`,
+			`git remote add boilerplate ${ this.BOILERPLATE_REPOSITORY }`,
+			`git fetch boilerplate ${ this.BOILERPLATE_BRANCH }`,
+			`git merge boilerplate/${ this.BOILERPLATE_BRANCH }`
+		];
+
+		tools.shExec( initializeCommands.join( ' && ' ) );
+	},
+
+	/**
+	 * Returns Git status of repository stored under specified path. It runs `git status --porcelain -sb` command.
+	 *
+	 * @param {String} repositoryPath Absolute path to repository.
+	 * @returns {String} Executed command's result.
+	 */
+	getStatus( repositoryPath ) {
+		return tools.shExec( `cd ${ repositoryPath } && git status --porcelain -sb` );
+	},
+
+	/**
+	 * Updates boilerplate project in specified repository.
+	 * @param {String} repositoryPath Absolute path to repository.
+	 */
+	updateBoilerplate( repositoryPath ) {
+		// Try to add boilerplate remote if one is not already added.
+		try {
+			tools.shExec( `cd ${ repositoryPath } && git remote add boilerplate ${ this.BOILERPLATE_REPOSITORY }` );
+		} catch ( error ) { }
```

Never ever catch to forget. It's better to let the whole execution blow up than to hide an error.